src/compiler/parser.ts

```diff
@@ -533,11 +534,12 @@ namespace ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode = false;
 
-        export function parseSourceFile(fileName: string, _sourceText: string, languageVersion: ScriptTarget, _syntaxCursor: IncrementalParser.SyntaxCursor, setParentNodes?: boolean): SourceFile {
-            const isJavaScriptFile = hasJavaScriptFileExtension(fileName) || _sourceText.lastIndexOf("// @language=javascript", 0) === 0;
+        export function parseSourceFile(fileName: string, _sourceText: string, languageVersion: ScriptTarget, _syntaxCursor: IncrementalParser.SyntaxCursor, setParentNodes?: boolean, scriptKind?: ScriptKind): SourceFile {
+
+            const isJavaScriptFile = scriptKind === ScriptKind.JS || hasJavaScriptFileExtension(fileName);

```

Comment:
I think it should be 
```
// just scriptKind as a condition handles both 
// - 'scriptKind' is unspecified and thus it is `undefined`
// - 'scriptKind' is set and it is `Unknown` (0)
const isJavaScriptFile = scriptKind 
    ? scriptKind === ScriptKind.JS || scriptKind === ScriptKind.JSX 
    : hasJavaScriptFileExtension(fileName)
```

otherwise case 'file extension is '.js'' and `scriptKind` is specified and is. `TS` will work incorrectly - left side of `||` will be evaluated to `false` and then final answer from the right side will be `true` whereis expected result (I think) should be `false`